core/src/main/java/com/novoda/noplayer/internal/exoplayer/PlayedAdverts.java

```diff
@@ -19,21 +20,36 @@ private PlayedAdverts() {
      * @param currentPositionInMillis The position before which all adverts will transition from their current state to Played.
      * @param advertBreaks            The client representation of the adverts, our source of truth.
      * @param adPlaybackState         The {@link AdPlaybackState} to alter advert state on.
-     * @return The {@link AdPlaybackState} with the newly played states.
+     * @return The {@link PlayedAdverts} instance with the newly played states.
      */
-    static AdPlaybackState markAllPastAdvertsAsPlayed(long currentPositionInMillis, List<AdvertBreak> advertBreaks, AdPlaybackState adPlaybackState) {
+    static PlayedAdverts from(long currentPositionInMillis, List<AdvertBreak> advertBreaks, AdPlaybackState adPlaybackState) {
```

Comment:
Can we go back to having the explicit name? `markAllPastAdvertsAsPlayed`? It still does the same from the looks of things 😄

Reply:
We discussed this offline (via slack, so it was technically online), now this is just a standard static factory method that creates an instance of `PlayedAdverts` so the name is updated to match other factory methods in the project.